swagger.yaml

```diff
@@ -2256,6 +2256,136 @@ paths:
           schema:
             $ref: '#/definitions/error'
 
+  /assisted-service-iso:
+    post:
+      tags:
+        - assisted-service-iso
+      summary: Creates ISO for the user and uploads to S3.
+      operationId: CreateISOAndUploadToS3
+      parameters:
+        - in: body
+          name: assisted-service-iso-create-params
+          required: true
+          schema:
+            $ref: '#/definitions/assisted-service-iso-create-params'
+      responses:
+        201:
+          description: Success.
+          schema:
+            $ref: '#/definitions/cluster'
```

Comment:
To align the functionality to generating cluster discovery ISO it should return an `ImageInfo` type, right?

Reply:
@jtomasek Since it's only 2 fields (pull secret and SSH key), and we get the pull secret from c.rh.com anyway, I thought maybe a DB is overkill.  Do you think it's necessary?

Reply:
@avishayt yes, the DB is not needed. The reason I'd expect the API call to return ImageInfo is for consistency with the discovery image generation. With the ImageInfo response (see below), the UI either uses `downloadUrl` (presigned url) to download, otherwise the UI uses `/assisted-service-iso/data` url to download the ISO.
```
type ImageInfo {
  /**
   * SSH public key for debugging the installation.
   */
  sshPublicKey?: string;
  sizeBytes?: number;
  downloadUrl?: string;
  /**
   * Image generator version.
   */
  generatorVersion?: string;
  createdAt?: string; // date-time
  expiresAt?: string; // date-time
}
```

In short, it would be good if POST /assisted-service-iso returned presigned download url as a response if it is available. With that the UI doesn't have to do another API call to find out if the presigned download url is available or not.

Reply:
Maintaining consistency by using ImageInfo may be difficult here. I believe ImageInfo is a Cluster field and a DB model. The assisted-service ISOs are not tied to a particular Cluster and their metadata isn't stored in the DB.

What we can do is, if the backend is AWS, create the presigned URL during POST /assisted-service-iso and return the presigned URL as a string response instead of ImageInfo.